### PR TITLE
EZP-31569: Repository filtering

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-31569
| Versions      | master (3.1)

### CODE REVIEW

File: docs/api/public_php_api_search.md [hunk 1]
```diff
@@ -81,10 +87,140 @@ As such, `query` is recommended when the search is based on user input.
...
+
+Notice that the total number of items is retrieved differently for `ContentList` and `LocationList`. 
+
+!!! caution
+
+    The total count is the total number of matched items, regardless of pagination settings.
+    
+!!! tip "Repository filtering is SiteAccess-aware"
+
+    Repository filtering is SiteAccess-aware, which means you can skip the second argument the `find` methods.
```
Comment (alongosz) [inline docs/api/public_php_api_search.md:180]: I think I've made a mistake here:
```suggestion
    Repository filtering is SiteAccess-aware, which means you can skip the second argument of the `find` methods.
```

File: docs/api/public_php_api_search.md [hunk 2]
```diff
@@ -81,10 +87,140 @@ As such, `query` is recommended when the search is based on user input.
...
+
+!!! note "Search Criteria and Sort Clause availability"
+
+    Not all Search Criteria and Sort Clauses are available for use in Repository filtering.
+    
+    Only Criteria implementing [`FilteringCriterion`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0/eZ/Publish/SPI/Repository/Values/Filter/FilteringCriterion.php)
+    and Sort Clauses implementing [`FilteringSortClause`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0/eZ/Publish/SPI/Repository/Values/Filter/FilteringSortClause.php)
+    are supported.
+
+    See to [Search Criteria](../guide/search/search_criteria_reference.md)
```
Comment (dabrt) [inline docs/api/public_php_api_search.md:212]:
```suggestion
    See [Search Criteria](../guide/search/search_criteria_reference.md)
```

File: docs/api/public_php_api_search.md [hunk 3]
```diff
@@ -81,10 +87,140 @@ As such, `query` is recommended when the search is based on user input.
...
+
+    It is recommended to use an IDE that can recognize type hints when working with Repository Filtering.
+    If you try to use an unsupported Criterion or Sort Clause, the IDE will indicate an issue. 
+
 ## Searching in a controller
 
-You can use the `SearchService` similarly in a controller, as long as you provide the required parameters.
-For example, in the code below `locationId` is provided to list all children of a Location.
+You can use the `SearchService` or Repository filtering in a controller, as long as you provide the required parameters.
+For example, in the code below `locationId` is provided to list all children of a Location using the `SearchService`.
```
Comment (dabrt) [inline docs/api/public_php_api_search.md:223]:
```suggestion
For example, in the code below, `locationId` is provided to list all children of a Location using the `SearchService`.
```
I guess "using" does not mean "that uses" but ""by using" or "with"? I'd remove the ambiguity.

File: docs/guide/search/search_criteria_reference.md [hunk 4]
```diff
@@ -16,47 +17,47 @@ Due to this storage limitation, searching content using the eZ Country Field Typ
...
-|[SectionId](criteria_reference/sectionid_criterion.md)|ID of the Section content is assigned to|Content and Location|
-|[SectionIdentifier](criteria_reference/sectionidentifier_criterion.md)|Identifier of the Section content is assigned to|Content and Location|
-|[Sibling](criteria_reference/sibling_criterion.md)|Locations that are children of the same parent|Content and Location|
-|[Subtree](criteria_reference/subtree_criterion.md)|Location subtree|Content and Location|
-|[UserEmail](criteria_reference/useremail_criterion.md)|Email address of a User account|Content and Location|
-|[UserId](criteria_reference/userid_criterion.md)|User ID|Content and Location|
-|[UserLogin](criteria_reference/userlogin_criterion.md)|User login|Content and Location|
-|[UserMetadata](criteria_reference/usermetadata_criterion.md)|The creator or modifier of a Content item|Content and Location|
-|[Visibility](criteria_reference/visibility_criterion.md)|Whether the Content item is visible or not|Content and Location|
+|[Ancestor](criteria_reference/ancestor_criterion.md)|Whether the Content item is an ancestor of the provided Location|Content and Location Search, and Filtering|
```
Comment (dabrt) [inline docs/guide/search/search_criteria_reference.md:22]:
```suggestion
|[Ancestor](criteria_reference/ancestor_criterion.md)|Whether the Content item is an ancestor of the provided Location|Content and Location, Search and Filtering|
```
WDYS? And below - "Location Search and Filtering", "Content and Location, Search"

File: docs/api/public_php_api_search.md (hunk 3 above)
Comment (DominikaK) [inline docs/api/public_php_api_search.md:223]:
```suggestion
For example, in the code below, `locationId` is provided to list all children of a Location by using the `SearchService`.
```

**Pull request merged**